### PR TITLE
[WIP] Make Handle validation erros for signups display

### LINKED ISSUE #64: Handle validation erros for signups
Currently if an user can not be created the user is stuck and we do not show any error message.

### PR DESCRIPTION
**Pull request by lewisdylan**
Displayed validation errors for signup on the welcome page
Closes #64

### CODE REVIEW

File: app/views/home/welcome.html.erb [hunk 1]
```diff
@@ -18,9 +18,13 @@
                                     Take the software development survey and get individual recommendations and 
                                     inspirations on how to improve your software development processes.
                                 </p>
-
+                              
                                 <div class="py-4 w-50">
-                                    <%= form_with url: api_users_path, model: User.new do |f| %>
+                            
+                                    <%= form_with url: api_users_path, model: @user ||= User.new do |f| %>
```
Comment (bumi) [inline app/views/home/welcome.html.erb]: don't assign an instance variable in the view. 

simply use `@user` and initialize that variable in the `welcome` action of the home controller

File: app/controllers/api/users_controller.rb [hunk 2]
```diff
@@ -9,12 +9,11 @@ def me
...
-      user = User.new(user_params)
-      if user.save && user.iterations.build.save # TODO: refactor iteration creation and redirect to iteration
-        redirect_to iteration_question_url(user_slug: user.slug, iteration_id: user.iterations.last.id, question_id: Question.first.id)
+      @user = User.new(user_params)
+      if @user.save && user.iterations.build.save # TODO: refactor iteration creation and redirect to iteration
+        redirect_to iteration_question_url(user_slug: @user.slug, iteration_id: @user.iterations.last.id, question_id: Question.first.id)
       else
-
-        render json: user.errors, status: :unprocessable_entity
+        render json: @user.errors, status: :unprocessable_entity
```
Comment (bumi) [inline app/controllers/api/users_controller.rb]: I am wondering about this. 

Why is the whole request an ajax request? can we make this a normal form post? 
Then we can render the page here and redirect in the success case (as we already do)

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -12,5 +12,7 @@ def index
   end
 
   # page rendered as root - when no user slug is given
-  def welcome; end
+  def welcome
+    @user ||= User.new do
```
Comment (bumi) [inline app/controllers/home_controller.rb]: this looks very much like a syntax error? - are you sure you tested this before? :)
also `@user` will never be defined here, so no need for `||=`

File: app/views/home/welcome.html.erb [hunk 4]
```diff
@@ -18,9 +18,13 @@
                                     Take the software development survey and get individual recommendations and 
                                     inspirations on how to improve your software development processes.
                                 </p>
-
+                              
                                 <div class="py-4 w-50">
-                                    <%= form_with url: api_users_path, model: User.new do |f| %>
+                            
+                                    <%= form_with url: api_users_path, model: @user |f| %>
```
Comment (bumi) [inline app/views/home/welcome.html.erb]: and here the `do` is missing - please always test what you are developing!

File: app/views/home/welcome.html.erb [hunk 5]
```diff
@@ -18,9 +18,13 @@
                                     Take the software development survey and get individual recommendations and 
                                     inspirations on how to improve your software development processes.
                                 </p>
-
+
```
Comment (bumi) [inline app/views/home/welcome.html.erb]: no need for this whitespace.

File: app/views/home/welcome.html.erb [hunk 6]
```diff
@@ -18,9 +18,13 @@
...
                                     inspirations on how to improve your software development processes.
                                 </p>
-
+                              
                                 <div class="py-4 w-50">
-                                    <%= form_with url: api_users_path, model: User.new do |f| %>
+                            
+                                    <%= form_with url: api_users_path, model: @user |f| %>
+                                       
+                                          <%= @user.errors.full_messages.first if @user.errors.any? %>
```
Comment (bumi) [inline app/views/home/welcome.html.erb]: you can display all of them, with something like:

```
<ul>
  <% @user.errors.full_messages.each do |msg| %>
    <li><%= msg %></li>
  <% end %>
</ul>
```

File: app/controllers/home_controller.rb (hunk 3 above)
Comment (musayann) [reply app/controllers/home_controller.rb]: Sorry my bad, I forgot to check if everything was working fine

File: app/views/home/welcome.html.erb (hunk 4 above)
Comment (musayann) [reply app/views/home/welcome.html.erb]: Sorry my bad, just fixed that

**Pull request merged**